src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,39 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User can`t be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User with this login already exists");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can`t be null");
+        }
+        if (user.getLogin().length() < MIN_LOGIN_LENGTH) {
+            throw new RegistrationException("Login length is less then 6 symbols");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password can`t be null");
+        }
+        if (user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new RegistrationException("Password length is less then 6 symbols");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age can`t be null");
+        }
+        if (user.getAge() < MIN_AGE) {
+            throw new RegistrationException("Only users over the age of 18 are allowed");
+        }
```

Comment:
I think you can decompose all this checks in separate functions for login, password and age for better readability